packages/react/src/navigation-menu/trigger/NavigationMenuTrigger.tsx

```diff
@@ -91,10 +95,32 @@ export const NavigationMenuTrigger = React.forwardRef(function NavigationMenuTri
   const [stickIfOpen, setStickIfOpen] = React.useState(true);
   const [pointerType, setPointerType] = React.useState<'mouse' | 'touch' | 'pen' | ''>('');
 
+  const allowFocusRef = React.useRef(false);
+  const prevSizeRef = React.useRef({ width: 0, height: 0 });
+  const abortControllerRef = React.useRef(new AbortController());
```

Comment:
It would be nice to put a bit more care into avoiding useless allocations. The size ref could have the empty/default state extracted to a static variable, and the abort controller ref could use `useLazyRef` to build only one `AbortController`.

Reply:
lint: The abort controller ref could be named for what it aborts, it would make it easier to understand what it does, e.g. `animationsController`.